src/Microsoft.VisualStudio.Composition/ExportProvider.cs

```diff
@@ -374,6 +374,73 @@ where importDefinition.ExportConstraints.All(c => c.IsSatisfiedBy(export.Definit
             return exportsSnapshot;
         }
 
+        /// <summary>
+        /// Releases the specified <see cref="Export"/> from the <see cref="ExportProvider"/>.
+        /// </summary>
+        /// <param name="export">The export to release.</param>
+        /// <remarks>
+        /// <para>When non-shared parts implement <see cref="IDisposable"/> they are referenced by this <see cref="ExportProvider"/>
+        /// until either the <see cref="ExportProvider"/> is disposed or the export is released by calling this method.</para>
+        /// <para>When provided an <see cref="Export"/> for a shared part, this method does nothing.</para>
+        /// </remarks>
+        public void ReleaseExport(Export export)
+        {
+            Requires.NotNull(export, nameof(export));
+
+            if (export is NonSharedExport disposable)
+            {
+                disposable.Dispose();
+            }
+        }
+
+        /// <inheritdoc cref="ReleaseExport(Export)"/>
+        public void ReleaseExport<T>(Lazy<T> export)
+        {
+            Requires.NotNull(export, nameof(export));
+
+            if (export is INonSharedLazy nonSharedExport)
```

Comment:
The Lazy itself isn't disposed, as it is not IDisposable. But we extract from the Lazy the Export, which we then get the PartLifecycleTracker from and dispose it. That Tracker disposes of the value that was returned from the `Lazy<T>`. Calling .Value after this point will throw if it hadn't been run earlier because the value factory we gave it can't produce a value after the export was disposed.